Source/drlg_l3.cpp

```diff
@@ -402,92 +402,77 @@ void CreateL3Dungeon(unsigned int rseed, int entry)
-
-				if (QuestStatus(QTYPE_ANVIL)) {
-					x2 = random(0, 10) + 10;
-					y2 = random(0, 10);
-					DRLG_L3FloorArea(x2, y2 + 10, x2 + 12, y2 + 22);
+				y1 = random(0, 20) + 10;
+				x2 = x1 + 2;
+				y2 = y1 + 2;
+				DRLG_L3FillRoom(x1, y1, x2, y2);
+				DRLG_L3CreateBlock(x1, y1, 2, 0);
+				DRLG_L3CreateBlock(x2, y1, 2, 1);
+				DRLG_L3CreateBlock(x1, y2, 2, 2);
+				DRLG_L3CreateBlock(x1, y1, 2, 3);
+				if(QuestStatus(QTYPE_ANVIL)) {
+					x1 = random(0, 10) + 10;
+					y1 = random(0, 10) + 10;
+					x2 = x1 + 12;
+					y2 = y1 + 12;
+					DRLG_L3FloorArea(x1, y1, x2, y2);
 				}
 				DRLG_L3FillDiags();
 				DRLG_L3FillSingles();
 				DRLG_L3FillStraights();
 				DRLG_L3FillDiags();
 				DRLG_L3Edges();
-				if (DRLG_L3GetFloorArea() < 600)
-					found = 0;
-				else
+				if(DRLG_L3GetFloorArea() >= 600) {
 					found = DRLG_L3Lockout();
-			} while (!found);
+				} else {
+					found = FALSE;
+				}
+			} while(!found);
 			DRLG_L3MakeMegas();
-			if (!entry) {
+			if(entry == 0) {
 				genok = DRLG_L3PlaceMiniSet(L3UP, 1, 1, -1, -1, 1, 0);
-				if (genok)
-					continue;
-				genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 0, 1);
-				if (genok)
-					continue;
-				if (currlevel != 9)
-					goto LABEL_24;
-				genok = DRLG_L3PlaceMiniSet(L3HOLDWARP, 1, 1, -1, -1, 0, 6);
-				goto LABEL_23;
-			}
-			genok = DRLG_L3PlaceMiniSet(L3UP, 1, 1, -1, -1, 0, 0);
-			if (entry == 1) {
-				if (genok)
-					continue;
-				genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 1, 1);
-				ViewX += 2;
-				ViewY -= 2;
-				if (genok)
-					continue;
-				if (currlevel != 9)
-					goto LABEL_24;
-				v24 = 0;
-			LABEL_22:
-				genok = DRLG_L3PlaceMiniSet(L3HOLDWARP, 1, 1, -1, -1, v24, 6);
-			LABEL_23:
-				if (genok)
-					continue;
-				goto LABEL_24;
+				if(!genok) {
+					genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 0, 1);
+					if(!genok && currlevel == 9) {
+						genok = DRLG_L3PlaceMiniSet(L3HOLDWARP, 1, 1, -1, -1, 0, 6);
+					}
+				}
+			} else if(entry == 1) {
+				genok = DRLG_L3PlaceMiniSet(L3UP, 1, 1, -1, -1, 0, 0);
+				if(!genok) {
+					genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 1, 1);
+					ViewX += 2;
+					ViewY -= 2;
+					if(!genok && currlevel == 9) {
+						genok = DRLG_L3PlaceMiniSet(L3HOLDWARP, 1, 1, -1, -1, 0, 6);
+					}
+				}
+			} else {
+				genok = DRLG_L3PlaceMiniSet(L3UP, 1, 1, -1, -1, 0, 0);
+				if(!genok) {
+					genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 0, 1);
+					if(!genok && currlevel == 9) {
+						genok = DRLG_L3PlaceMiniSet(L3HOLDWARP, 1, 1, -1, -1, 1, 6);
+					}
+				}
 			}
-			if (genok)
-				continue;
-			genok = DRLG_L3PlaceMiniSet(L3DOWN, 1, 1, -1, -1, 0, 1);
-			if (genok)
-				continue;
-			if (currlevel == 9) {
-				v24 = 1;
-				goto LABEL_22;
+			if(!genok && QuestStatus(QTYPE_ANVIL)) {
+				genok = DRLG_L3Anvil();
 			}
-		LABEL_24:
-			if (!QuestStatus(QTYPE_ANVIL))
-				break;
-			genok = DRLG_L3Anvil();
-		} while (genok == 1);
+		} while(genok == TRUE);

```

Comment:
```diff
- 		} while(genok == TRUE); 
+ 		} while(genok); 
```

Reply:
While that may be how *we* write code, that unfortunately is not the same thing. It is `!= 0` rather than `== 1` which has different asm ;)

Reply:
Aah, fair point.